src/VisualStudio/CSharp/Impl/CodeModel/CSharpCodeModelService.cs

```diff
@@ -3021,6 +3021,14 @@ protected override ITypeSymbol GetTypeSymbolFromPartialName(string partialName,
 
         public override ITypeSymbol GetTypeSymbolFromFullName(string fullName, Compilation compilation)
         {
+            // Some clients pass us an assembly qualified name, which we don't handle.  Just lop off the
+            // assembly info if so.
+            var index = fullName.IndexOf(',');
+            if (index > 0)
+            {
+                fullName = fullName.Substring(0, index);
+            }
+
             ITypeSymbol typeSymbol = compilation.GetTypeByMetadataName(fullName);
 
             if (typeSymbol == null)
```

Comment:
@DustinCampbell what are the cases where you expect this to be null?  I believe the symptoms in the bug would be because `GetTypeByMetadataName` doesn't handle the assembly qualification and then returns null, so we hit this code path, but `SyntaxFactory.ParseTypeName` only consumes up to the `+`, so we end up with a `TypeSyntax` that doesn't include any nested types.

Is this fallback code actually necessary for other cases (possibly actually language specific generics, keyword types, etc?), or should it be deleted now?